src/coreclr/src/jit/morph.cpp

```diff
@@ -7783,6 +7783,12 @@ GenTree* Compiler::fgMorphTailCallViaHelpers(GenTreeCall* call, CORINFO_TAILCALL
         call->gtFlags &= ~GTF_CALL_VIRT_STUB;
```

Comment:
We do it for `stub` and in my understanding for `vtable` we should do the same, let's see what ci thinks.

Reply:
This is `fgMorphTailCallViaHelpers` which is for the new mechanism. Indeed those turn into normal direct calls always.

Reply:
it looks like I will need the same fix for `fgMorphTailCallViaJitHelper`: **https://github.com/sandreenko/runtime/commit/53cea4e4038ad5b3821a7a6b3734364f71ec7514** , but I want to see if our ci catches it.

Reply:
It looks like we remove all virtual kinds below: https://github.com/dotnet/runtime/blob/dc3101bc11eeef183b6d7c3ab38f0a13248c7783/src/coreclr/src/jit/morph.cpp#L7917-L7921

I guess this line is not required but doesn't hurt since this is where we remove the arg for virtual stubs.

Reply:
I see, thank you. I removed changes in `fgMorphTailCallViaHelpers` and fixed `fgMorphTailCallViaJitHelper` in the same way.